### PR TITLE
Consens-o-meter!

### PR DESCRIPTION
**Pull request by brandonfancher**
### No votes cast
image

### No clear leader (2 vs. 2)
image

### Leader emerged but not enough votes yet
image

### Enough votes but leader not voting for self
image

### Consensus!
image

### Unanimous consensus
image

### CODE REVIEW

File: packages/webapp/src/elections/api/fixtures.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export const fixtureVoteDataRows: VoteData[] = [
         member: "egeon.edev",
         round: 1,
         index: 5,
-        candidate: "edenmember13",
```
Comment (GreenBusDriver) [inline packages/webapp/src/elections/api/fixtures.ts:90]: Do you want to leave this scenario in the fixture? fine if so; just wanted to ensure you intended to alter the fixture in this PR

File: packages/webapp/src/elections/components/ongoing-election-components/consensometer.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    const {
+        isThereConsensus,
+        leadCandidates,
+        leaderIsVotingForSelf,
+        leadTally,
+        totalVotesCast,
+        totalVotesRequiredForConsensus,
+        remainingVotesRequiredForConsensus,
+    } = tallyVotesFromVoteData(voteData);
```
Comment (GreenBusDriver) [inline packages/webapp/src/elections/components/ongoing-election-components/consensometer.tsx:55]: love this

File: packages/webapp/src/elections/utils.ts [hunk 3]
```diff
@@ -24,3 +26,51 @@ export const extractElectionDates = (election: any) => {
...
+    return {
+        totalVotesCast: votes.length,
+        leadCandidates: candidatesWithMostVotes,
+        leadTally,
+        totalVotesRequiredForConsensus: threshold,
+        remainingVotesRequiredForConsensus: threshold - leadTally,
+        leaderIsVotingForSelf,
+        isThereConsensus: didReachConsensus,
+    };
+};
```
Comment (GreenBusDriver) [inline packages/webapp/src/elections/utils.ts:76]: some solid JS ninjaing in here. :)

**Review by GreenBusDriver — APPROVED** [review]

File: packages/webapp/src/elections/api/fixtures.ts (hunk 1 above)
Comment (brandonfancher) [reply packages/webapp/src/elections/api/fixtures.ts:90]: Yeah, the fixture had `egeon.edev` voting for someone outside their round.

**Review by cool-ant — APPROVED** [review]

**Pull request merged**